examples/http-client-oauth2-client-credentials-grant-type/http_client_oauth2_client_credentials_grant_type.md

```diff
@@ -1,6 +1,6 @@
 # HTTP client - OAuth2 client credentials grant type
 
-A client, which is secured with OAuth2 client credentials grant type can be used to connect to a secured service. The client is enriched with the `Authorization: Bearer <token>` header by passing the `http:OAuth2ClientCredentialsGrantConfig` for the `auth` configuration of the client.
+An `http:Client` can be secured with OAuth2 client credentials grant type authorization. This is achieved by enriching each request from the client with the `Authorization: Bearer <token>` header. The required configurations for the OAuth2 client credentials grant type can be specified in the `auth` field as the `http:OAuth2ClientCredentialsGrantConfig` record. Use this to communicate with the service, which is secured with OAuth2 authorization.
```

Comment:
We could add the following to answer **When to use ?** question
Use when client application needs to access resources or services on behalf of itself rather than on behalf of a user.

Reply:
Added via [187d99d](https://github.com/ballerina-platform/ballerina-distribution/pull/3793/commits/187d99dffcd52f92e019e05f224ac4dc5e4a2e66)